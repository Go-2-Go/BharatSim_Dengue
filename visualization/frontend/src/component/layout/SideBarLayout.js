import React from 'react';
import PropTypes from 'prop-types';
import { Box, Typography } from '@material-ui/core';

import styles from './sideBarLayoutCSS';

import bharatSimLogo from '../../assets/images/Logobharatsim-logo.svg';

import labels from './sidebarLayout.labels';

function SideBarLayout({ ControllerComponent, ViewComponent }) {
  const classes = styles();

  return (
    <Box className={classes.main}>
      <Box className={classes.sideBar}>
        <Box className={classes.logo}>
          <Box pr={5}>
            <img src={bharatSimLogo} alt="logo" />
          </Box>
          <Typography variant="h1" className={classes.logoText}>
            {labels.LOGO_TEXT}
          </Typography>
        </Box>
        <Box>
          <ControllerComponent />
        </Box>
      </Box>
      <Box className={classes.view}>
        <ViewComponent />
      </Box>
    </Box>
  );
}

SideBarLayout.propTypes = {
  ControllerComponent: PropTypes.elementType.isRequired,
  ViewComponent: PropTypes.elementType.isRequired,
};

export default SideBarLayout;
