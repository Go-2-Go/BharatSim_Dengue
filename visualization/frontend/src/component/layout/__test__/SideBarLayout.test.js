import React from 'react';
import { render } from '@testing-library/react';
import SideBarLayout from '../SideBarLayout';

function DummyController() {
  return <div>DummyController</div>;
}

function DummyView() {
  return <div>DummyView</div>;
}

describe('<SideBarLayout />', () => {
  it('should mactch snapshot', () => {
    const { container } = render(
      <SideBarLayout ViewComponent={DummyView} ControllerComponent={DummyController} />,
    );

    expect(container).toMatchSnapshot();
  });
});
