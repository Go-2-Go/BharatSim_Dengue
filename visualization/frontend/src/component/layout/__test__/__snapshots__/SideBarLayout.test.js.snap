// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBarLayout /> should mactch snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6 makeStyles-main-1"
  >
    <div
      class="MuiBox-root MuiBox-root-7 makeStyles-sideBar-2"
    >
      <div
        class="MuiBox-root MuiBox-root-8 makeStyles-logo-4"
      >
        <div
          class="MuiBox-root MuiBox-root-9"
        >
          <img
            alt="logo"
            src="Logobharatsim-logo.svg"
          />
        </div>
        <h1
          class="MuiTypography-root makeStyles-logoText-5 MuiTypography-h1"
        >
          BHARATSIM
        </h1>
      </div>
      <div
        class="MuiBox-root MuiBox-root-10"
      >
        <div>
          DummyController
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-11 makeStyles-view-3"
    >
      <div>
        DummyView
      </div>
    </div>
  </div>
</div>
`;
